### PR TITLE
[Doc][Iceberg] Improved iceberg documentation

### PR DESCRIPTION
**Pull request by Carl-Zhou-CN**
subtask of #4572 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/source/Iceberg.md [hunk 1]
```diff
@@ -22,126 +28,112 @@ Source connector for Apache Iceberg. It can support batch and stream mode.
...
 ```
-schema {
-    fields {
-      f2 = "boolean"
-      f1 = "bigint"
-      f3 = "int"
-      f4 = "bigint"
-    }
-}
+flink-shaded-hadoop-x-xxx.jar
```
Comment (zhilinli123) [inline docs/en/connector-v2/source/Iceberg.md:48]: https://github.com/apache/seatunnel/blob/dev/docs/en/connector-v2/source/Mysql.md
image
Refer to the current documentation to add a jar dependent address boot for the user

Comment (Carl-Zhou-CN) [reply docs/en/connector-v2/source/Iceberg.md:48]: done

**Comment by zhilinli123** [comment]
good , Thanks for your contribution. Looks good  PTAL : @hailin0

**Review by Hisoka-X — APPROVED** [review]

**Review by zhilinli123 — APPROVED** [review]

**Review by liugddx — APPROVED** [review]

**Comment by zengqinchris** [comment]
When can use the function of sink to iceberg

**Pull request merged**